tsconfig.settings.json

```diff
@@ -0,0 +1,28 @@
+{
+    "extends": "./node_modules/@salesforce/dev-config/tsconfig",
+    "compilerOptions": {
+      "module": "commonjs",
+      "target": "es6",
+      "lib": ["es6", "dom"],
+      "sourceMap": true,
+      "declaration": true,
+      "moduleResolution": "node",
+      "noImplicitAny": true,
+      "outDir": "./lib",
+      "preserveConstEnums": true,
+      "strict": true,
+      "allowSyntheticDefaultImports": true,
+      "skipLibCheck": true,
+  
+      // We can replace the TS generated snippets with require statements
+      // from the tslib dependency, making our code more efficient
+      "importHelpers": true,
+      "resolveJsonModule": true
+  
+      // Commenting composite out since it prevents the debugger from working
+      // if we import metadatainfos.json
+      // "composite": true
+    },
+    "include": ["./src/**/*"]
+  }
+  
```

Comment:
Is there a reason all of this ts config is in its own file and not part of `tsconfig.json`? ([tsconfig.settings.json](https://github.com/salesforcecli/plugin-templates/pull/1/files#diff-6d9dfe806b0aeffa01011c28e580110501b307dc617ec8f214dc1c6986dbcdf4))

Reply:
Historically, there probably was, but that knowledge is long lost now 😄 I'll try moving it and see if it's as simple as that!